src/cephadm/cephadm

```diff
@@ -4363,24 +4363,51 @@ def command_bootstrap(ctx):
     if ctx.apply_spec:
         logger.info('Applying %s to cluster' % ctx.apply_spec)
 
+        # process spec file for host specs
+        host_specs = []
         with open(ctx.apply_spec) as f:
+            temp_host_spec = {}  # type: Dict[str, str]
+            host_spec_bool = False
             for line in f:
-                if 'hostname:' in line:
-                    line = line.replace('\n', '')
-                    split = line.split(': ')
-                    if split[1] != hostname:
-                        logger.info('Adding ssh key to %s' % split[1])
-
-                        ssh_key = '/etc/ceph/ceph.pub'
-                        if ctx.ssh_public_key:
-                            ssh_key = ctx.ssh_public_key.name
-                        out, err, code = call_throws(ctx, ['sudo', '-u', ctx.ssh_user, 'ssh-copy-id', '-f', '-i', ssh_key, '-o StrictHostKeyChecking=no', '%s@%s' % (ctx.ssh_user, split[1])])
+                if '---' in line:
+                    if temp_host_spec:
+                        host_specs.append(temp_host_spec)
+                    temp_host_spec = {}
+                    host_spec_bool = False
+                elif host_spec_bool:
+                    if 'hostname:' in line or 'addr:' in line:
+                        line = line.replace('\n', '')
+                        split = line.split(': ')
+                        temp_host_spec[split[0]] = split[1]
+                elif not host_spec_bool:
+                    if 'service_type: host' in line:
+                        host_spec_bool = True
+            if temp_host_spec:
+                host_specs.append(temp_host_spec)
+
+        ssh_key = '/etc/ceph/ceph.pub'
+        if ctx.ssh_public_key:
+            ssh_key = ctx.ssh_public_key.name
+
+        # copy ssh key for host specs
+        for spec in host_specs:
+            if 'addr' in spec:
+                addr = spec['addr']
+            else:
+                addr = spec['hostname']
+            out, err, code = call(ctx, ['sudo', '-u', ctx.ssh_user, 'ssh-copy-id', '-f', '-i', ssh_key, '-o StrictHostKeyChecking=no', '%s@%s' % (ctx.ssh_user, addr)])
+            if code:
+                logger.info('\nCopying ssh key to  %s failed!\n' % addr)
```

Comment:
I think it would be nice to have the hostname of the host incorporated into the error message as well (even if it is the same as the address). Something like
```suggestion
                logger.info('\nCopying ssh key to host %s at address %s failed!\n' % (spec['hostname'], addr))
```